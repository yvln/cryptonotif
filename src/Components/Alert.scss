.Alert {
  background-color: rgba(38, 70, 83, .8);
  color: white;
  margin: 10px;
  width: 180px;
  min-height: 110px;
}

.AlertText {
  padding: 20px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 100%;
  max-width: 140px;
  min-height: 90px;
  position: relative;

  .buttons {
    display: flex;
    justify-content: space-between;
    align-items: flex-start;

    > button {
      border: none;
      background-color: transparent;
      font-size: 15px;

      &:hover {
        font-size: 17px;
      }
    }
  }

  > div {
    text-align: center;
    margin-bottom: 10px;
  }
}

.AlertForm {
  background-color: rgb(38, 70, 83);
  position: absolute;
  width: 100%;
  max-width: 180px;
  height: 240px;
  z-index: 2;

  transform: translateY(-10px);
}
