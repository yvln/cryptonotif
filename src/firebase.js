/* eslint-disable */

const firebase = require('firebase');
require('dotenv').config();

const config = {
  apiKey: process.env.API_KEY,
  authDomain: 'cryptonotif-56a23.firebaseapp.com',
  databaseURL: 'https://cryptonotif-56a23.firebaseio.com',
  projectId: 'cryptonotif-56a23',
  storageBucket: 'cryptonotif-56a23.appspot.com',
  messagingSenderId: '1008924569797',
};

export const initializeFirebase = () => {
  firebase.initializeApp(config);
  return true;
};

export const getDatabase = () => {
  return firebase.database();
};
