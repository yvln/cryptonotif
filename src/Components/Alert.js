import React, { Component } from "react";

import { getDatabase } from "../firebase";
import Form from "./Form";
// import "./Alert.css";

class Alert extends Component {
  state = {
    isEditing: false
  };

  closeForm = () => {
    this.setState({
      isEditing: false
    });
  };

  closeAlert = () => {
    getDatabase()
      .ref(`/alert/${this.props.data.key}`)
      .remove();
  };

  editAlert = () => {
    this.setState({
      isEditing: true
    });
  };

  render() {
    console.log(this.props.data);
    const { data } = this.props;

    if (!data) {
      return "Loading...";
    }

    return (
      <div className="Alert">
        {this.state.isEditing ? (
          <>
            <div onClick={this.closeForm}>Cancel</div>
            <Form
              initialValues={{
                data
              }}
              closeForm={this.closeForm}
            />
          </>
        ) : (
          <>
            <div onClick={this.closeAlert}>close</div>
            <div onClick={this.editAlert}>edit</div>
            {data.action} {data.amount} {data.asset}
          </>
        )}
      </div>
    );
  }
}

export default Alert;
