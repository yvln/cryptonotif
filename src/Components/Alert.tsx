import React, { Component } from 'react';

import { getDatabase } from '../firebase';
import Form from './Form';
import './Alert.scss';

type AlertProps = {
  data: {
    action: string;
    amount: string;
    asset: string;
    currency: string;
    email: string;
    key: string;
  };
  id: number;
  key: string;
};

type AlertState = {
  current: string | undefined;
  isEditing: boolean;
};

class Alert extends Component<AlertProps, AlertState> {
  state = {
    current: undefined,
    isEditing: false,
  };

  componentDidMount() {
    getDatabase()
      .ref(`/current`)
      .on('value', snapshot => {
        if (!snapshot) {
          return;
        }

        this.setState({
          current: snapshot.val()[this.props.data.asset],
        });
      });
  }

  closeForm = () => {
    this.setState({
      isEditing: false,
    });
  };

  closeAlert = () => {
    getDatabase()
      .ref(`/alert/${this.props.data.key}`)
      .remove();
  };

  editAlert = () => {
    this.setState({
      isEditing: true,
    });
  };

  render() {
    const { data } = this.props;

    if (!data) {
      return 'Loading...';
    }

    return (
      <div className="Alert">
        {this.state.isEditing ? (
          <div className={this.state.isEditing && 'AlertForm'}>
            <button onClick={this.closeForm}>Cancel</button>
            <Form
              initialValues={{
                data,
              }}
              closeForm={this.closeForm}
            />
          </div>
        ) : (
          <div className={!this.state.isEditing && 'AlertText'}>
            <div className="buttons">
              <button onClick={this.editAlert}>
                <span role="img" aria-label="edit">
                  üñäÔ∏è
                </span>
              </button>
              <button onClick={this.closeAlert}>
                <span role="img" aria-label="close">
                  ‚ùå
                </span>
              </button>
            </div>
            <div>
              When {data.asset} is {data.action} {data.amount}
              {data.currency} to: {data.email}
              {this.state.current && <div>CURRENT: {this.state.current}</div>}
            </div>
          </div>
        )}
      </div>
    );
  }
}

export default Alert;
