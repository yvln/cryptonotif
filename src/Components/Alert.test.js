import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import renderer from 'react-test-renderer';

import { initializeFirebase } from '../firebase';

import Alert from './Alert';

configure({ adapter: new Adapter() });

describe('src/Components/Alert', () => {
  it('should render', () => {
    initializeFirebase();
    const component = renderer.create(
      <Alert
        data={{
          asset: 'BTC',
          amount: '600',
          action: 'above',
          currency: 'USD',
          email: 'itsme@contact.com',
        }}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it("should display 'Loading' when no data props", () => {
    const alert = shallow(<Alert />);
    expect(alert.text()).toEqual('Loading...');
  });

  it('should display an alert when receiving data props', () => {
    const alert = mount(
      <Alert
        data={{
          asset: 'BTC',
          amount: '600',
          action: 'above',
          currency: 'USD',
          email: 'itsme@contact.com',
        }}
      />
    );
    expect(alert.text()).toContain('When BTC is above 600USD');
  });
});
