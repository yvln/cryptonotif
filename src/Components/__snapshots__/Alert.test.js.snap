// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/Components/Alert should render 1`] = `
<div
  className="Alert"
>
  <div
    className="AlertText"
  >
    <div
      className="buttons"
    >
      <button
        onClick={[Function]}
      >
        <span
          aria-label="edit"
          role="img"
        >
          üñäÔ∏è
        </span>
      </button>
      <button
        onClick={[Function]}
      >
        <span
          aria-label="close"
          role="img"
        >
          ‚ùå
        </span>
      </button>
    </div>
    <div>
      When 
      BTC
       is 
      above
       
      600
      USD
       to: 
      itsme@contact.com
    </div>
  </div>
</div>
`;
